src/sfctl/custom_cluster.py

```diff
@@ -78,6 +81,12 @@ def select(endpoint, cert=None, key=None, pem=None, ca=None,
 
         # Make sure basic GET request succeeds
         rest_client.send(rest_client.get('/')).raise_for_status()
+
+        # Get cluster version
+        cluster_code_versions = rest_client.send(
```

Comment:
This assumes that there has been a baseline upgrade performed, otherwise this will be an empty list. It's also bad practice to assume this URL. You should use a client and use the installed Service Fabric SDK to make this API call. Take a look at how `get_aad_token` works, I believe that's a good example.